File: main/src/main/java/com/sedmelluq/discord/lavaplayer/tools/http/MultiHttpConfigurable.java
package com.sedmelluq.discord.lavaplayer.tools.http;

import org.apache.http.client.config.RequestConfig;
import org.apache.http.impl.client.HttpClientBuilder;

import java.util.Collection;
import java.util.function.Consumer;
import java.util.function.Function;

public class MultiHttpConfigurable implements ExtendedHttpConfigurable {
    private final Collection<ExtendedHttpConfigurable> configurables;

    public MultiHttpConfigurable(Collection<ExtendedHttpConfigurable> configurables) {
        this.configurables = configurables;
    }

    @Override
    public void setHttpContextFilter(HttpContextFilter filter) {
        for (ExtendedHttpConfigurable configurable : configurables) {
            configurable.setHttpContextFilter(filter);
        }
    }

    @Override
    public void configureRequests(Function<RequestConfig, RequestConfig> configurator) {
        for (ExtendedHttpConfigurable configurable : configurables) {
            configurable.configureRequests(configurator);
        }
    }

    @Override
    public void configureBuilder(Consumer<HttpClientBuilder> configurator) {
        for (ExtendedHttpConfigurable configurable : configurables) {
            configurable.configureBuilder(configurator);
        }
    }
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/tools/http/MultiHttpConfigurable.kt
package com.sedmelluq.discord.lavaplayer.tools.http

import com.sedmelluq.discord.lavaplayer.tools.io.BuilderConfigurator
import com.sedmelluq.discord.lavaplayer.tools.io.RequestConfigurator

open class MultiHttpConfigurable(private val configurables: Collection<ExtendedHttpConfigurable>) :
    ExtendedHttpConfigurable {
    override fun setHttpContextFilter(filter: HttpContextFilter) {
        for (configurable in configurables) {
            configurable.setHttpContextFilter(filter)
        }
    }

    override fun configureRequests(configurator: RequestConfigurator) {
        for (configurable in configurables) {
            configurable.configureRequests(configurator)
        }
    }

    override fun configureBuilder(configurator: BuilderConfigurator) {
        for (configurable in configurables) {
            configurable.configureBuilder(configurator)
        }
    }
}
